msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenMockTest.java

```diff
@@ -0,0 +1,577 @@
+            }
+
+        }.performTest();
+    }
+
+    @Test
+    public void canAcquireSilentAfterGettingToken() {
+        new AcquireTokenBaseTest() {
+
+            @Override
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+
+                String username = "fake@test.com";
+
+                final AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+                        .withScopes(Arrays.asList(SCOPES))
+                        .forceRefresh(false)
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                        .startAuthorizationFromActivity(activity)
+                        .withLoginHint(username)
+                        .withScopes(Arrays.asList(SCOPES))
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                                IAccount account = authenticationResult.getAccount();
+                                silentParameters.setAccount(account);
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                publicClientApplication.acquireToken(parameters);
+                RoboTestUtils.flushScheduler();
+                publicClientApplication.acquireTokenSilentAsync(silentParameters);
```

Comment:
we might have to put this inside acquireToken's callback?

Reply:
That would have been the ideal case but Robolectric's threading model does not allow us to do that. By doing it this way, we are basically treating these as synchronous methods.